target/linux/lantiq/modules.mk

```diff
@@ -17,3 +17,21 @@ endef
 
 $(eval $(call KernelPackage,i2c-lantiq))
 
+define KernelPackage/dsa-gswip
```

Comment:
It should also work when you compile the driver into the kernel. If the FW is not available it will send a probe deferred and try again later.

Reply:
It looks like deferring doesn't work:
~~~
...
[    0.730599] 0x0000000c0000-0x000000100000 : "unused"
[    0.736344] 0x000000100000-0x000007f80000 : "ubi"
[    0.969652] gswip 1e108000.switch: Direct firmware load for lantiq/xrx200_phy11g_a22.bin failed with error -2
[    0.978157] gswip 1e108000.switch: Falling back to sysfs fallback for: lantiq/xrx200_phy11g_a22.bin
[   64.539678] gswip 1e108000.switch: error -ETIMEDOUT: failed to load firmware: lantiq/xrx200_phy11g_a22.bin
[   64.547936] gswip 1e108000.switch: error -ETIMEDOUT: gphy fw probe failed
[   64.554785] gswip: probe of 1e108000.switch failed with error -145
...
~~~
Maybe something changed between 5.15 and 6.6.

Reply:
ok, I am sure it was working some years ago. I was using this when developing the driver.